src/tree/node.rs

```diff
@@ -0,0 +1,384 @@
+use std::cmp::Ordering;
+use std::cmp::Ordering::*;
+use std::mem::{self, replace, swap};
+
+use compare::Compare;
+
+// Nodes keep track of their level in the tree, starting at 1 in the
+// leaves and with a red child sharing the level of the parent.
+#[derive(Clone)]
+pub struct Node<K, V> {
+    key: K,
+    value: V,
+    left: Option<Box<Node<K, V>>>,
+    right: Option<Box<Node<K, V>>>,
+    level: usize,
```

Comment:
Since Node has two generic parameters, using u8 has the chance of netting some benefit. Rust can (theoretically) reorder the fields to get a better packing. u8 will sometimes be a wash with u32, but sometimes better (e.g. if K = V = u8).

However this is a TBD optimization for the compiler. CC @huonw